substrate/frame/revive/src/precompiles/builtin/system.rs

```diff
@@ -44,6 +44,8 @@ impl<T: Config> BuiltinPrecompile for System<T> {
 	) -> Result<Vec<u8>, Error> {
 		use ISystem::ISystemCalls;
 		match input {
+			ISystemCalls::terminate(ISystem::terminateCall { .. }) if env.is_read_only() =>
+				Err(LibError::<T>::StateChangeDenied.into()),
```

Comment:
`LibError` is inconsistent and not anywhere used like that. It also doesn't provide additional context (of course everything is a library - the alias would be more justified if it would be e.g. `PalletError` or something similar).

```suggestion
				Err(crate::Error::<T>::StateChangeDenied.into()),
```